drivers/pwm/pwm_renesas_ra8.c

```diff
@@ -0,0 +1,584 @@
+	if (err != FSP_SUCCESS) {
+		return -EIO;
+	}
+	*cycles = (uint64_t)info.clock_frequency;
+
+	return 0;
+};
+
+#ifdef CONFIG_PWM_CAPTURE
+extern void gpt_capture_compare_a_isr(void);
+extern void gpt_counter_overflow_isr(void);
+
+static void enable_irq(IRQn_Type const irq, uint32_t priority, void *p_context)
+{
+	if (irq >= 0) {
+		R_BSP_IrqCfgEnable(irq, priority, p_context);
+	}
+}
+static void disable_irq(IRQn_Type irq)
+{
+	/* Disable interrupts. */
+	if (irq >= 0) {
+		R_BSP_IrqDisable(irq);
+		R_FSP_IsrContextSet(irq, NULL);
+	}
+}
+
+static int pwm_ra8_configure_capture(const struct device *dev, uint32_t pin, pwm_flags_t flags,
+				     pwm_capture_callback_handler_t cb, void *user_data)
+{
+	struct pwm_ra8_data *data = dev->data;
+
+	if (pin != GPT_IO_PIN_GTIOCA) {
+		LOG_ERR("Feature only support for gtioca");
+		return -EINVAL;
+	}
+	if (!(flags & PWM_CAPTURE_TYPE_MASK)) {
+		LOG_ERR("No PWWM capture type specified");
+		return -EINVAL;
+	}
+	if ((flags & PWM_CAPTURE_TYPE_MASK) == PWM_CAPTURE_TYPE_BOTH) {
+		LOG_ERR("Cannot capture both period and pulse width");
+		return -ENOTSUP;
+	}
+	if (data->capture.is_busy) {
+		LOG_ERR("Capture already active on this pin");
+		return -EBUSY;
+	}
+
+	if (flags & PWM_CAPTURE_TYPE_PERIOD) {
+		data->capture.is_pulse_capture = false;
+
+		if (flags & PWM_POLARITY_INVERTED) {
+			data->extend_cfg.start_source =
+				(gpt_source_t)(GPT_SOURCE_GTIOCA_FALLING_WHILE_GTIOCB_LOW |
+					       GPT_SOURCE_GTIOCA_FALLING_WHILE_GTIOCB_HIGH |
+					       GPT_SOURCE_NONE);
+			data->extend_cfg.capture_a_source = data->extend_cfg.start_source;
+
+		} else {
+			data->extend_cfg.start_source =
+				(gpt_source_t)(GPT_SOURCE_GTIOCA_RISING_WHILE_GTIOCB_LOW |
+					       GPT_SOURCE_GTIOCA_RISING_WHILE_GTIOCB_HIGH |
+					       GPT_SOURCE_NONE);
+			data->extend_cfg.capture_a_source = data->extend_cfg.start_source;
+		}
+	} else {
+		data->capture.is_pulse_capture = true;
+
+		if (flags & PWM_POLARITY_INVERTED) {
+			data->extend_cfg.start_source =
+				(gpt_source_t)(GPT_SOURCE_GTIOCA_FALLING_WHILE_GTIOCB_LOW |
+					       GPT_SOURCE_GTIOCA_FALLING_WHILE_GTIOCB_HIGH |
+					       GPT_SOURCE_NONE);
+
+			data->extend_cfg.capture_a_source =
+				(gpt_source_t)(GPT_SOURCE_GTIOCA_RISING_WHILE_GTIOCB_LOW |
+					       GPT_SOURCE_GTIOCA_RISING_WHILE_GTIOCB_HIGH |
+					       GPT_SOURCE_NONE);
+		} else {
+			data->extend_cfg.start_source =
+				(gpt_source_t)(GPT_SOURCE_GTIOCA_RISING_WHILE_GTIOCB_LOW |
+					       GPT_SOURCE_GTIOCA_RISING_WHILE_GTIOCB_HIGH |
+					       GPT_SOURCE_NONE);
+
+			data->extend_cfg.capture_a_source =
+				(gpt_source_t)(GPT_SOURCE_GTIOCA_FALLING_WHILE_GTIOCB_LOW |
+					       GPT_SOURCE_GTIOCA_FALLING_WHILE_GTIOCB_HIGH |
+					       GPT_SOURCE_NONE);
+		}
+	}
+
+	data->capture.callback = cb;
+	data->capture.user_data = user_data;
+	data->capture.continuous = (flags & PWM_CAPTURE_MODE_CONTINUOUS) ? true : false;
+
+	if (data->capture.continuous == true) {
```

Comment:
`if (data->capture.continuous) {`

it's a boolean...

Reply:
I updated it, thanks